src/serac/numerics/petsc_solvers.cpp

```diff
@@ -5,6 +5,10 @@
 // SPDX-License-Identifier: (BSD-3-Clause)
 
 #include "serac/numerics/petsc_solvers.hpp"
+
+#include <cstddef>  // for NULL
```

Comment:
Two points:

* I personally dislike comments that tell you what each header is for
* Why are we using `NULL` and not `nullptr`?